parsing/lexer.mll

```diff
@@ -785,6 +852,14 @@ and quoted_string delim = parse
         if delim = edelim then lexbuf.lex_start_p
         else (store_lexeme lexbuf; quoted_string delim lexbuf)
       }
+  | "|" (ident_ext as raw_edelim) "}"
```

Comment:
Is this optimisation necessary ?
Also, as `delim` has been validated before entering this state, the `None` case is not necessary as the `Error` value returned by `UIdent.normalize` cannot be equal `delim`. This could removes duplicated code.